manifests/init.pp

```diff
@@ -169,4 +170,10 @@
     }
   }
 
+  if $repo_scl {
+    ::rhsm::repo { "rhel-server-rhscl-${::operatingsystemmajrelease}-rpms":
```

Comment:
mirrored/cloned channels provided by redhat  satellite or other tools might not use this repo name. 

what if redhat decides to change channel names as they did for example with oracle jre/jdk channels? 

I think the boolean param thing is not a good way to provide options for these additional channels. 

the rhsm module should not try to guess repo names.

Reply:
I agree, but is there a way to get the same result, using a manifest or hiera, without having to change the module?

Reply:
@ubellavance I don't know your puppet setup. if you go with the roles/profiles pattern:


```
class profiles::rhsm {
  include ::rhsm
  rhsm::repo { "rhel-server-rhscl-${::operatingsystemmajrelease}-rpms"
}
```

if you need to pass config by hiera data a new `$repos` rhsm class parameter might be good.

```
class rhsm(
  ...
  Optional[Array[String]] $repos = undef,
  ...
) {
  ...
  repos.each do |String repo| {
    ::rhsm::repo { $repo: }
  }
}
```

as per https://github.com/voxpupuli/puppet-rhsm/issues/66#issuecomment-383889760 - the code depends on if puppet3 server needs to be supported or not (this code is not puppet3 server compatible)

if the `rhsm::repo` supports params, then the `$repos` param maybe should not be an array but an hash with support for the params of rhsm::repo.